plugin/evm/vm.go

```diff
@@ -1071,55 +1071,39 @@ func (vm *VM) initBlockBuilding() error {
 	var (
 		ethTxGossipHandler    p2p.Handler
 		atomicTxGossipHandler p2p.Handler
+		ethTxGossiper         gossip.Gossiper
+		atomicTxGossiper      gossip.Gossiper
 	)
 
-	ethTxGossipHandler, err = gossip.NewHandler[*GossipEthTx](ethTxPool, ethTxGossipHandlerConfig, vm.sdkMetrics)
+	ethTxGossipClient, err := vm.Network.NewClient(ethTxGossipProtocol, p2p.WithValidatorSampling(vm.validators))
 	if err != nil {
-		return err
-	}
-	ethTxGossipHandler = &p2p.ValidatorHandler{
-		ValidatorSet: vm.validators,
-		Handler: &p2p.ThrottlerHandler{
-			Handler:   ethTxGossipHandler,
-			Throttler: p2p.NewSlidingWindowThrottler(throttlingPeriod, throttlingLimit),
-			Log:       vm.ctx.Log,
-		},
-		Log: vm.ctx.Log,
-	}
-
-	if err := vm.Network.AddHandler(ethTxGossipProtocol, ethTxGossipHandler); err != nil {
-		return err
+		return fmt.Errorf("failed to create eth tx gossip client: %w", err)
 	}
 
-	atomicTxGossipHandler, err = gossip.NewHandler[*GossipAtomicTx](vm.mempool, atomicTxGossipHandlerConfig, vm.sdkMetrics)
+	ethTxGossipHandler, err = gossip.NewHandler[GossipEthTx, *GossipEthTx](
+		vm.ctx.Log,
+		ethTxGossipClient,
+		ethTxPool,
+		ethTxGossipHandlerConfig,
+		vm.sdkMetrics,
+	)
 	if err != nil {
 		return err
 	}
-
-	atomicTxGossipHandler = &p2p.ValidatorHandler{
+	ethTxGossipHandler = &p2p.ValidatorHandler{
```

Comment:
these reassignments seems a bit complicated. can we use a different naming for these or try to create a separate `New...` function?